# Parameters for the SLAM node.

# Frequency at which we want to operate the SLAM ROS node.
node_loop_rate: 50

# Topic in which the IMU measurements are being received.
imu_meas_topic: /imu/data

# Measurement noise characteristics:
# --Prior factors:
# Standard deviation of the prior information's translation components [m].
prior_pos_stddev: 0.0001
# Standard deviation of the prior information's rotation components [deg].
prior_rot_stddev: 0.000001

# Sonar Extrinsics
# NOTE: In the IMU frame, relative pose from IMU to the sonar. IMU in ENU
# configuration, sonar Z axis pointing down, Y right, X forward. [3D + quat]
ext_x: -0.02484
ext_y: 0.14525
ext_z: -1.22172
quat_x: 0.93477
quat_y: -0.06936
quat_z: -0.27065
quat_w: -0.21936

# --IMU measurements:
# VALUES FOR SBG ELLIPSE A
# Standard deviation for the linear acceleration measurements [m/s2].
#accel_noise_x: 5.6898e-3
#accel_noise_y: 5.7879e-3
#accel_noise_z: 5.1993e-3
accel_noise_x: 5.6e-4
accel_noise_y: 5.6e-4
accel_noise_z: 5.6e-4
# Bias standard deviation for the linear acceleration measurements [m/s2]
#accel_bias_x: 4.8069e-3
#accel_bias_y: 3.0411e-3
#accel_bias_z: 3.7278e-3
accel_bias_x: 1.37e-4
accel_bias_y: 1.37e-4
accel_bias_z: 1.37e-4
# NOTE: Values depend on the IMU frequency!!
# Standard deviation for the angular velocity measurements [???].
#gyro_noise_x: 1.047197e-5
#gyro_noise_y: 1.047197e-5
#gyro_noise_z: 1.047197e-5
gyro_noise_x: 3.085e-4
gyro_noise_y: 3.085e-4
gyro_noise_z: 3.085e-4
# Bias standard deviation for the angular velocity measurements [rad/s]
gyro_bias_x: 3.3936e-5
gyro_bias_y: 3.3936e-5
gyro_bias_z: 3.3936e-5
# Measurement frequency [Hz].
imu_frequency: 50.0

# SLAM estimate publishers.
dead_reckoning_topic: /slam/dead_reckoning/path
dead_reckoning_frame_id: world
